src/main/java/Deadline.java

```diff
@@ -0,0 +1,20 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = LocalDate.parse(by);
+    }
+
+    @Override
+    public String toTXT() { return "D | " + super.toTXT() + " | " + by; }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by:" + by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```

Comment:
This string operation is a bit confusing. Would you consider using `String.format(...` or splitting it into multiple lines?